content/articles/client-side-authentication-with-express-cookie-parser/index.md

```diff
@@ -0,0 +1,329 @@
+- An understanding of a JavaScript programming language.
+- A pre-installed IDE, preferably [Visual Studio Code](https://code.visualstudio.com/download).
+- [Node](https://nodejs.org/en//) installed.
+- An understanding of [Express](https://expressjs.com/).
+
+
+### Objectives
+In this tutorial, we will be learning the client-side authentication with web browser cookies using the express cookies parser library. In addition, you will learn the following and apply them to the project;
+- Overview of web browser cookies.
+- Browser Cookies Properties.
+- Demo of client-side authentication with cookies parser library
+- Making browser cookies secured from attacks
+
+
+### Overview of web browser cookies
+The web browser cookies are found on the console page. Cookies are a medium of storage that is used for varieties of things like client authentication, setting of language, tracking information, and more.
+
+There are two main ways of setting the browser cookies. Both the client-side and server are useful when it comes to saving the cookies. Specifically, we want the browser to always remember a bit of information, and that is why cookies are mainly useful.
+
+In addition, cookies are set in the browsers whenever there is a request, where `set-cookie` gets attached to the `headers` like the `Content-type`. This `set-cookie` has the property of `key-value` pairs where `key` represents the `name`, and `value` equals the cookie `data` to be set.
+
+Furthermore, let us demonstrate how to set a cookie at the client-side by navigating to the domain called `example.com`. On this domain page, right-click anywhere, and open to the console.
+
+You can set the cookie at the client side of this domain by typing the snippet below in the console.
+
+```JavaScript
+document.cookie="example=domain"
+```
+
+The `example` in the snippet above serves as the `key`, while the `domain` is the `value`. Note that, this can be anything of your choice. The image below is the description of the cookie setting.
+
+![Example Cookie](/engineering-education/client-side-auth-with-express-cookies-parser/cookie.png)
+
+
+
+### Browser cookies properties
+You need to know that cookies are sent on every client request to the server. Every domain name serves as a bucket that any cookie created can be sent to. For instance, when we sent a request to the domain `example.com` earlier, no cookie came with it. But we later injected one through the console and that filled the `name`, and `value` properties. 
+
+Browser cookies have different properties that developers can make use. Every attribute can be filled either through client injection in the console or the server whenever there is a request to the backend. Let us go through the feature of cookies together.
+
+##### Name
+This is the name given to the particular cookie saved in the browser. This can be filed through the `key` in the cookie data.
+
+##### Value
```

Comment:
@Horlawhumy-dev Please use h4 instead of h5 for these headings.

Reply:
That is #### right?